qiskit/cluster.py

```diff
@@ -7,30 +7,45 @@
 
 class Processor:
 
-    def __init__(self, cluster, index, nr_qubits):
+    def __init__(self, cluster, index, nr_qubits,flag):
         self.cluster = cluster
+        self.flag=flag
         self.qc = cluster.qc
         self.index = index
         names = ['alice', 'bob', 'charlie', 'david', 'eve', 'frank', 'george', 'harry']
         if index > len(names):
             self.name = str(index)
         else:
             self.name = names[index]
-        self.main_reg = QuantumRegister(nr_qubits, f'{self.name}_main')
-        self.qc.add_register(self.main_reg)
-        self.teleport_reg = QuantumRegister(1, f'{self.name}_teleport')
-        self.qc.add_register(self.teleport_reg)
-        self.entanglement_reg = QuantumRegister(1, f'{self.name}_entanglement')
-        self.qc.add_register(self.entanglement_reg)
-        self.measure_reg = ClassicalRegister(2, f'{self.name}_measure')
-        self.qc.add_register(self.measure_reg)
+
+        if flag == "teleport":
+            self.main_reg = QuantumRegister(nr_qubits, f'{self.name}_main')
+            self.qc.add_register(self.main_reg)
+            self.teleport_reg = QuantumRegister(1, f'{self.name}_teleport')
+            self.qc.add_register(self.teleport_reg)
+            self.entanglement_reg = QuantumRegister(1, f'{self.name}_entanglement')
+            self.qc.add_register(self.entanglement_reg)
+            self.measure_reg = ClassicalRegister(2, f'{self.name}_measure')
+            self.qc.add_register(self.measure_reg)
+
+        if flag == "cat":
+
+            # note: teleport_reg has been removed
```

Comment:
We could the common part (the main register, the entanglement register, and the classical register) outside the if statement instead of copy-pasting it in both methods.

Reply:
Agreed